File: airesources/CSharp/Halite2/hlt/Log.cs
using System.IO;

namespace Halite2.hlt
{
    public class Log
    {
        private TextWriter file;
        private static Log instance;

        private Log(TextWriter f)
        {
            file = f;
        }

        public static void Initialize(TextWriter f)
        {
            instance = new Log(f);
        }

        public static void LogMessage(string message)
        {
            try
            {
                instance.file.WriteLine(message);
                instance.file.Flush();
            }
            catch (IOException)
            {
            }
        }
    }
}

File: airesources/Kotlin/src/kotlin/halite/Log.kt
package halite

import java.io.FileWriter
import java.io.IOException

class Log private constructor(private val file: FileWriter) {
    companion object {
        private var instance: Log? = null

        internal fun initialize(f: FileWriter) {
            instance = Log(f)
        }

        fun log(message: String) {
            try {
                instance!!.file.write(message)
                instance!!.file.write("\n")
                instance!!.file.flush()
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
    }
}
